sat/cli/bootsys/platform.py

```diff
@@ -347,45 +347,49 @@ def run(self):
                 with code 1. If it's successful, raises a SystemExit with a code 0.
                 This is used to stop the process.
         """
-        exit_status, stdout, stderr = self._run_remote_command('systemctl is-active containerd',
-                                                               err_on_non_zero=False)
-        if exit_status:
-            if 'inactive' in stdout:
-                self._success_exit(f'containerd is not active on {self.host} so '
-                                   f'there are no containers to stop.')
+        while True:
+            exit_status, stdout, stderr = self._run_remote_command('systemctl is-active containerd',
+                                                                   err_on_non_zero=False)
+            if exit_status:
+                if 'inactive' in stdout:
+                    self._success_exit(f'containerd is not active on {self.host} so '
+                                       f'there are no containers to stop.')
+                else:
+                    self._err_exit(f'Failed to query if containerd is active. '
+                                   f'stdout: {stdout}, stderr: {stderr}')
+
+            running_containers = self._get_running_containers()
+
+            if not running_containers:
+                self._success_exit(f'No containers to stop on {self.host}.')
+
+            if not self.success:
+                LOGGER.debug('The following containers were running before the stop attempt '
+                             'on %s: %s', self.host, running_containers)
+
+            exit_status, stdout, stderr = self._run_remote_command(CONTAINER_STOP_SCRIPT,
+                                                                   err_on_non_zero=False)
+
+            stopped_containers = stdout.splitlines()
+            if not self.success:
+                LOGGER.debug('The following containers were stopped successfully on %s: %s',
+                             self.host, stopped_containers)
+
+            if exit_status:
+                # This is most likely a timeout but not necessarily an error if
+                # containers have been stopped.
+                LOGGER.warning(f'One or more "crictl stop" commands timed out on {self.host}')
+
+            # Check and see if they've all been stopped.
+            running_containers = self._get_running_containers()
+            self.ssh_client.close()
+
+            if running_containers:
+                LOGGER.warning(f'Failed to stop containers on {self.host}: {running_containers}')
+                LOGGER.info(f'Retrying container stop procedure on {self.host}')
+                time.sleep(10)  # Wait for a while before retrying
```

Comment:
I'm not sure if it's necessary to sleep between attempts.

Reply:
Added sleep time to speedup the process, if not necessary will reduce to 1 sec before next attempt